File: backend/app/schemas/workflow.py
from pydantic import BaseModel


class WorkflowRunRequest(BaseModel):
    workflow_key: str
    inputs: dict


class WorkflowRunResponse(BaseModel):
    workflow_run_id: str
    status: str
    outputs: dict | None = None
    error: str | None = None


class ChatRequest(BaseModel):
    query: str
    conversation_id: str = ""


class ChatResponse(BaseModel):
    answer: str
    conversation_id: str
    sources: list[dict] = []

File: frontend/src/api/workflow.ts
import request from './request'

interface WorkflowResponse {
  workflow_run_id: string
  status: string
  outputs?: Record<string, any>
  error?: string
}

interface ChatResponse {
  answer: string
  conversation_id: string
  sources: Record<string, any>[]
}

export function runWorkflow(data: { workflow_key: string; inputs: Record<string, any> }) {
  return request.post<WorkflowResponse>('/workflow/run', data)
}

export function chatWithKB(data: { query: string; conversation_id?: string }) {
  return request.post<ChatResponse>('/workflow/chat', data)
}
